lib/problems.rb

```diff
@@ -1,13 +1,40 @@
-require_relative './stack.rb'
+require_relative './Stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(nlogn)
+# Space Complexity: O(n)
 def balanced(string)
```

Comment:
Since you have to go through all the chars and you're not splitting at each stage, this is O(n) for time complexity.